parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -2785,6 +2785,12 @@ opt_block_args_tail:
                         lambda->end);
                       DIAGCHECK();
                     }
+                | t_lambda error
+                    {
+                      driver.lex.unextend();
+                      driver.lex.context.inLambda = $1->context.inLambda;
+                      YYERROR;
+                    }
```

Comment:
Heads up, this doesn't affect your change, but I wanted to make sure that it didn't introduce conflicts, and I remembered that the only way to know that was to check the build logs for warnings.

Now, after #8270 lands, that will be enforced in CI. I've already checked that this new `error` production of yours builds successfully when rebased on my change.

Reply:
I rebuilt with those flags locally to make sure, and it passed. Thanks for adding those!